clippy_lints/src/loops/manual_find.rs

```diff
@@ -0,0 +1,145 @@
+        if let Some(last_ret) = last_ret(cx, expr);
+        then {
+            let mut applicability = Applicability::MachineApplicable;
+            let mut snippet = make_iterator_snippet(cx, arg, &mut applicability);
+            let is_ref_to_binding =
+                matches!(pat.kind, PatKind::Ref(inner, _) if matches!(inner.kind, PatKind::Binding(..)));
+            if !(matches!(pat.kind, PatKind::Binding(..)) || is_ref_to_binding) {
+                snippet.push_str(
+                    &format!(
+                        ".map(|{}| {})",
+                        snippet_with_applicability(cx, pat.span, "..", &mut applicability),
+                        snippet_with_applicability(cx, inner_ret.span, "..", &mut applicability),
+                    )[..],
+                );
+            }
+            let mut needs_copy = false;
+            match cond.kind {
+                ExprKind::Binary(_, l, r) => {
+                    for side in [l, r] {
+                        if !matches!(
+                            side.kind,
+                            ExprKind::MethodCall(_, [caller, ..], _) if is_local_used(cx, caller, binding_id)
+                        ) && is_local_used(cx, side, binding_id)
+                        {
+                            needs_copy = true;
+                        }
+                    }
+                },
+                ExprKind::Call(..) => needs_copy = true,
+                ExprKind::MethodCall(_, [caller, ..], _) => {
+                    if !is_local_used(cx, caller, binding_id) {
+                        needs_copy = true;
+                    }
+                },
+                _ => {},
+            }
+            let ty = cx.typeck_results().expr_ty(inner_ret);
+            if !needs_copy || cx.tcx.lang_items().copy_trait().map_or(false, |id| implements_trait(cx, ty, id, &[])) {
+                snippet.push_str(
+                    &format!(
+                        ".find(|{}{}| {})",
+                        "&".repeat(usize::from(needs_copy) + usize::from(is_ref_to_binding)),
+                        snippet_with_applicability(cx, inner_ret.span, "..", &mut applicability),
+                        snippet_with_applicability(cx, cond.span, "..", &mut applicability),
+                    )[..],
+                );
+                if is_ref_to_binding {
+                    snippet.push_str(".copied()");
+                }
+            } else {
+                applicability = Applicability::HasPlaceholders;
+                snippet.push_str(
+                    &format!(
+                        ".find(|{}| {{ ... }})",
+                        snippet_with_applicability(cx, inner_ret.span, "..", &mut applicability),
+                    )[..],
+                );
+            }
+            span_lint_and_sugg(
+                cx,
+                MANUAL_FIND,
+                span.to(last_ret.span),
+                "manual implementation of Iterator::find",
```

Comment:
```suggestion
                "manual implementation of `Iterator::find`",
```

code snippets should be in backticks.